src/unittest_json.nim

```diff
@@ -97,18 +97,19 @@ method testEnded(formatter: JsonOutputFormatter, testResult: TestResult) =
     if formatter.testStackTrace.len > 0:
       jsonTestResult = JsonTestResult(
         name: testResult.testName,
-        status: FAIL,
+        status: ERROR,
         message: fmt"{failureMsg}\n{formatter.testStackTrace}"
       )
       if errs.len > 0:
         jsonTestResult.message.insert(errs & "\n")
+      formatter.result = ResultJson(status: ERROR, tests: formatter.result.tests)
```

Comment:
```suggestion
      formatter.result = ResultJson(status: FAIL, tests: formatter.result.tests)
```
My understanding is that the `ERROR` status is only for when the test can't run. Not if there is an unexpected error thrown.

Reply:
You're right, thanks. Fixed.

I think we can make the specification clearer on that point.

@iHiD quote from Slack:
> If the solution fails to compile or the tests can’t run for some reason, then the top level status should be `error`. If the tests can run, then the top level status should be `pass` or `fail`. So if a test runs but `error`s, that should result in the top level status being `fail`.

> To put it another way, in the UI, if we get a top level `error`, we’ll only show the top level message, not any of the test fails/errors.